engine/class_modules/sc_death_knight.cpp

```diff
@@ -6139,6 +6137,24 @@ struct scourge_strike_base_t : public death_knight_melee_attack_t
   int n_targets() const override
   { return p() -> in_death_and_decay() ? as<int>( p() -> spec.scourge_strike -> effectN( 4 ).base_value() ) : 0; }
 
+  std::vector<player_t*>& target_list() const override // smart targeting to targets with wounds when cleaving SS 
```

Comment:
Does it always prioritize higher stack counts or just targets with wounds over targets that don't have wounds?

Reply:
I've been told it just picks wounds over no wounds but had issues getting the sort lambda to work when I tried to model that behavior.  Tried setting two bools based on if the targets had wounds and would return the first bool if it was true otherwise return the second.  This threw exceptions in the middle of execution, saying the lambda wasn't returning a bool.  After a couple of hours of fiddling with it I stripped it down to just return the result of comparison between the two wound counts.  Though even that first solution was imperfect because it favored targets that came earlier in the target list.  The most technically correct solution would be to make a vector of n_targets minus the first with wounds and then use rng to pull 4 random ones out, and then assemble a new target list of the action targer, the 4 randomly chosen targets, then the rest.  Hmm, might have just rubber ducked my way to an answer.

Reply:
What happens if you just replace the a_stacks and b_stacks value by 0 if the debuff isn't up, and 1 if the buff is up?
Something like 
```   int a_stacks = td( a )->debuff.festering_wound->up() ? 1 : 0;
      int b_stacks = td( b )->debuff.festering_wound->up() ? 1 : 0;
      return a_stacks > b_stacks; ```
If your current implementation works I don't think there should be an issue with simplifying stack count to just 0s and 1s.

Reply:
Ah that's a good idea.  That's a more straightforward way of what I tried to do on the first attempt.  It will still favor targets that appear first in the list but will be easier to understand and maintain than trying to randomly pick up to 4 extra targets with wounds and I'm not sure that the outcomes would be different if we did it that way.

Reply:
That did the job nicely https://pastebin.pl/view/raw/751b175b